src/Symfony/Component/VarDumper/Dumper/CliDumper.php

```diff
@@ -610,19 +610,30 @@ private function hasColorSupport($stream): bool
             return false;
         }
 
-        if ('Hyper' === getenv('TERM_PROGRAM')) {
+        // Detect msysgit/mingw and assume this is a tty because detection
+        // does not work correctly, see https://github.com/composer/composer/issues/9690
+        if (!@stream_isatty($stream) && !\in_array(strtoupper((string) getenv('MSYSTEM')), ['MINGW32', 'MINGW64'], true)) {
             return true;
```

Comment:
The logic in this method does not match the logic in the console component. A stream that is not a TTY must not report supporting colors.